### ISSUE TITLE
Why were the internal modules hidden?

### ISSUE THREAD

**Issue by emilaxelsson** [sdiehl/protolude#64]
Prior to version 0.2 I was able to make my own adapted version of protolude by just copying `Protolude.hs` and make some changes to it. This adapted protolude still depended on the protolude package to get access to the other modules. That's no longer possible since the internal modules are not exported.

In fact, I was under the impression that this way of customizing Protolude.hs was the intended use of the package, since it's said to be a "Prelude template".

Any chance the modules could be exported again?

**Comment by sdiehl** [comment]
All of the datatypes and functions should be exported from `Protolude.hs` so you should be able pick and choose what you want from the reexporting any number of symbols from this module. Is there something missing at the moment?

**Comment by emilaxelsson** [comment]
OK, but say that I want to hide some stuff exported by Protolude.hs. Previously I could just copy the file and comment out the unwanted modules, whereas now I'll have to hide individual identifiers explicitly.

**Comment by sdiehl** [comment]
If you want to mask one of the modules provided, then [removing](https://github.com/sdiehl/protolude/blob/master/src/Protolude.hs#L35) one of these lines should be enough.

To understand better, are you forking the repo for a custom project or using protolude itself as library?

**Comment by emilaxelsson** [comment]
I'm not forking the repo, just copying Protolude.hs (to a different name) and depending on the protolude package to get access to the other modules.

And the modules I want to hide are not the ones included in the protolude package -- only external modules that are reexported by Protolude.hs.

But I see now that I can just replace these lines

```haskell
import List as X
import Show as X
...
```

with

```haskell
import Protolude as X
```

in my copy of Protolude.hs. That should be enough for me!

Thanks for getting me on the right track :)

**Comment by emilaxelsson** [comment]
No, sorry, I got confused. I cannot say

```
import Protolude as X
```

in my copy of Protolude.hs, because then I'll get everything, including the external re-exports that I want to hide.

So I think version 0.2 makes it harder to use protolude as a library upon which you can build your own custom prelude.

**Comment by sdiehl** [comment]
Ahh, I see so you are importing protolude in your cabal file and then copying `Protolude.hs` over to construct a custom Prelude module in the same way that this library does. 

This makes sense, I just haven't seen anyone else use this library like that (most people just fork the whole project).

If you like I can migrate the internal modules over to an a seperate namespace `Internal.List`, `Internal.Show` and then this should support your use case without conflicting with other namespaces (such as GHC's itself).

**Comment by emilaxelsson** [comment]
Yes, that would be nice! Maybe the namespace `Protolude.List`, etc. would be even less likely to clash with other modules (but it doesn't matter to me).

**Comment by sdiehl** [comment]
This is now fixed in 9533e739ff8fa951766414450dca3d66e9c4b942. Let me know if this is optimal for your use case.

**Comment by emilaxelsson** [comment]
That's excellent, thank you!
